File: Daily Challanges/2026-06-June/2026-06-25.py
def parse_frontmatter(frontmatter_string: str) -> dict:
    parsed_data = {}
    lines = frontmatter_string.strip().split("\n")

    for line in lines:
        if line == "---" or not line.strip():
            continue

        if ": " in line:
            key, value_str = line.split(": ", 1)

            if value_str == "true":
                parsed_data[key] = True
            elif value_str == "false":
                parsed_data[key] = False
            else:
                try:
                    parsed_data[key] = int(value_str)
                except ValueError:
                    try:
                        parsed_data[key] = float(value_str)
                    except ValueError:
                        parsed_data[key] = value_str

    return parsed_data


print("Test 1:")
test_1 = parse_frontmatter("---\ntitle: My Post\ndraft: false\nviews: 100\n---")
print(test_1)
print(f"Passed: {test_1 == {'title': 'My Post', 'draft': False, 'views': 100}}\n")

print("Test 2:")
test_2 = parse_frontmatter(
    "---\nid: 6a174db57256a112f932195c\ntitle: My Book\nlocale: en\nwordCount: 10000\npublished: false\n---"
)
print(test_2)
print(
    f"Passed: {test_2 == {'id': '6a174db57256a112f932195c', 'title': 'My Book', 'locale': 'en', 'wordCount': 10000, 'published': False}}\n"
)

print("Test 3:")
test_3 = parse_frontmatter(
    "---\nversion: 1.0.0\nurl: https://example.com\nprivate: true\n---"
)
print(test_3)
print(
    f"Passed: {test_3 == {'version': '1.0.0', 'url': 'https://example.com', 'private': True}}\n"
)

print("Test 4:")
test_4 = parse_frontmatter("---\nrating: 4.5\nprice: 9.99\n---")
print(test_4)
print(f"Passed: {test_4 == {'rating': 4.5, 'price': 9.99}}")

File: Daily Challanges/2026-06-June/2026-06-25.js
function parseFrontmatter(frontmatterString) {
  const parsedData = {};
  const lines = frontmatterString.trim().split("\n");

  for (const line of lines) {
    if (line === "---" || !line.trim()) {
      continue;
    }

    if (line.includes(": ")) {
      const splitIndex = line.indexOf(": ");
      const key = line.slice(0, splitIndex);
      const valueStr = line.slice(splitIndex + 2);

      if (valueStr === "true") {
        parsedData[key] = true;
      } else if (valueStr === "false") {
        parsedData[key] = false;
      } else {
        const num = Number(valueStr);
        if (!isNaN(num) && valueStr.trim() !== "") {
          parsedData[key] = num;
        } else {
          parsedData[key] = valueStr;
        }
      }
    }
  }

  return parsedData;
}

console.log("Test 1:");
const test1 = parseFrontmatter(
  "---\ntitle: My Post\ndraft: false\nviews: 100\n---",
);
console.log(test1);

console.log("\nTest 2:");
const test2 = parseFrontmatter(
  "---\nid: 6a174db57256a112f932195c\ntitle: My Book\nlocale: en\nwordCount: 10000\npublished: false\n---",
);
console.log(test2);

console.log("\nTest 3:");
const test3 = parseFrontmatter(
  "---\nversion: 1.0.0\nurl: https://example.com\nprivate: true\n---",
);
console.log(test3);

console.log("\nTest 4:");
const test4 = parseFrontmatter("---\nrating: 4.5\nprice: 9.99\n---");
console.log(test4);
